includes/EntityValidateBase.php

```
@@ -129,204 +99,114 @@ public function getFieldsInfo() {
-        $this->isValidValue($value, $field, $field_type_info['property_type']);
+        $this->isValidValue($field_name, $wrapper->{$field_name}->value(), $field_type_info['property_type']);
       }
 
       if (!empty($info['validators'])) {
-        $info['validators'] = array_unique($info['validators']);
-        foreach ($info['validators'] as $validator) {
-          $value = $wrapper->__isset($field) ? $wrapper->{$field}->value() : $entity->{$field};
-          call_user_func_array($validator, array($value, $field));
-        }
+        $this->invokeMethods($wrapper->{$field_name}, $info['validators']);
       }
     }
 
     // Throwing exception with the errors.
-    if (!empty($this->errors)) {
+    if (!$silent && !empty($this->errors)) {
       $params = array(
         '@errors' => implode(", ", $this->errors),
       );
 
       throw new \EntityValidatorException(t('The validation process failed: @errors', $params));
     }
-
-    return TRUE;
-  }
-
-  /**
-   * {@inheritdoc}
-   */
-  public function setError($message) {
-    $this->errors[] = $message;
-  }
-
-  /**
-   * Verify the field is not empty.
-   *
-   * @param $value
-   *  The value of the field.
-   * @param $field
-   *  The field name.
-   *
-   * @return boolean
-   */
-  public function isNotEmpty($value, $field) {
-    if (empty($value)) {
-      $params = array(
-        '@field' => $field,
-      );
-
-      $this->setError(t("The field @field can't be empty", $params));
-    }
-  }
-
-  /**
-   * Check if the field is a text field.
-   *
-   * @param $value
-   *  The value of the field.
-   * @param $field
-   *  The field name.
-   *
-   * @return boolean
-   */
-  public function isText($value, $field) {
-    if (!is_string($value)) {
-      $params = array(
-        '@value' => $value,
-      );
-
-      $this->setError('The given value(@value) is not a string', $params);
-      return;
+    else {
+      return $this->errors;
```

I don't think we need to return the errors, just TRUE or FALSE. later implementing module can `$this->getErrors()`